packages/block-library/src/gallery/edit.js

```diff
@@ -69,148 +66,202 @@ const MOBILE_CONTROL_PROPS_RANGE_CONTROL = Platform.select( {
 	native: { type: 'stepper' },
 } );
 
+const DEFAULT_GUTTER_SIZE = 16;
+const MAX_GUTTER_SIZE = 100;
+const MIN_GUTTER_SIZE = 0;
+
 function GalleryEdit( props ) {
 	const {
+		setAttributes,
 		attributes,
+		className,
+		clientId,
+		noticeOperations,
 		isSelected,
 		noticeUI,
-		noticeOperations,
-		mediaUpload,
-		imageSizes,
-		resizedImages,
-		onFocus,
+		insertBlocksAfter,
 	} = props;
+
 	const {
-		columns = defaultColumnsNumber( attributes ),
+		imageCount,
+		columns = defaultColumnsNumber( imageCount ),
+		gutterSize,
 		imageCrop,
-		images,
+		imageUploads,
+		linkTarget,
 		linkTo,
+		shortCodeTransforms,
 		sizeSlug,
 	} = attributes;
-	const [ selectedImage, setSelectedImage ] = useState();
-	const [ attachmentCaptions, setAttachmentCaptions ] = useState();
-	const { __unstableMarkNextChangeAsNotPersistent } = useDispatch(
-		'core/block-editor'
-	);
 
-	function setAttributes( newAttrs ) {
-		if ( newAttrs.ids ) {
-			throw new Error(
-				'The "ids" attribute should not be changed directly. It is managed automatically when "images" attribute changes'
-			);
-		}
-
-		if ( newAttrs.images ) {
-			newAttrs = {
-				...newAttrs,
-				// Unlike images[ n ].id which is a string, always ensure the
-				// ids array contains numbers as per its attribute type.
-				ids: map( newAttrs.images, ( { id } ) => parseInt( id, 10 ) ),
-			};
-		}
-
-		props.setAttributes( newAttrs );
-	}
-
-	function onSelectImage( index ) {
-		return () => {
-			setSelectedImage( index );
+	const {
+		__unstableMarkNextChangeAsNotPersistent,
+		replaceInnerBlocks,
+	} = useDispatch( 'core/block-editor' );
+
+	const { getSettings, preferredStyle } = useSelect( ( select ) => {
+		const settings = select( 'core/block-editor' ).getSettings();
+		const preferredStyleVariations =
+			settings.__experimentalPreferredStyleVariations;
+		return {
+			getBlock: select( 'core/block-editor' ).getBlock,
+			getSettings: select( 'core/block-editor' ).getSettings,
+			preferredStyle: preferredStyleVariations?.value?.[ 'core/image' ],
 		};
-	}
+	}, [] );
 
-	function onDeselectImage() {
-		return () => {
-			setSelectedImage();
-		};
-	}
+	const innerBlockImages = useSelect( ( select ) => {
+		return select( 'core/block-editor' ).getBlock( clientId )?.innerBlocks;
+	} );
```

Comment:
This is missing `[ clientId ]` as second argument which is causing some kind of "loop" where running the selector retriggers a rendering which it self retriggers the same selector.

Reply:
Thanks @youknowriad that explains a lot  👍

I can confirm with that fixed, no errors are thrown when images are reordered or new ones dropped into the gallery etc.